tests/components/dyson/test_sensor.py

```diff
@@ -137,6 +190,83 @@ def test_dyson_filter_life_sensor_with_values(self):
         assert sensor.entity_id == "sensor.dyson_1"
         sensor.on_message("message")
 
+    def test_dyson_hepa_filter_state_sensor(self):
```

Comment:
For the new tests, please instead write the tests like this as standalone async test functions and use the core interfaces:
https://github.com/home-assistant/core/blob/aecc3476e84fcdd87675c381c9eea01247f2a338/tests/components/dyson/test_climate.py#L213-L244

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
Hah, it turns out it's actually easier to test it this way, and it tests the one Combi selection logic trick too. Thanks!